src/batch/src/task/task_execution.rs

```diff
@@ -500,41 +495,51 @@ impl<C: BatchTaskContext> BatchTaskExecution<C> {
             tokio::select! {
                 // We prioritize abort signal over normal data chunks.
                 biased;
-                err_reason = &mut shutdown_rx => {
-                    match err_reason {
-                        Ok(reason_str) => {
-                            state = TaskStatus::Aborted;
-                            error = Some(Aborted(reason_str));
-                        }
+                res = shutdown_rx.changed() => {
```

Comment:
Do we still need this? What about making it the responsibility of executors and propagating the error here?

Reply:
There is one possibility happen:
The executor done or last poll and shutdown(e.g. user cancel) happen in concurency. In this case we may miss the shutdown and consider the execution as successful.🤔 

But seems it's ok to consider the execution as successful so we can remove this.

Reply:
Yes. I guess it doesn't matter if we treat this task as succeeded or failed in this case.